File: app/src/main/java/com/rankin/adam/cookingmaster/Model/RecipeList.java
package com.rankin.adam.cookingmaster.Model;

import java.lang.reflect.Array;
import java.util.ArrayList;

/**
 * Created by Adam on 14-Jul-18.
 */

public class RecipeList {

    private ArrayList<Recipe> recipeList;

    public RecipeList() {
        this.recipeList = new ArrayList<>();
    }

    public void addRecipe(Recipe recipe){
        recipeList.add(recipe);
    }

    public void editRecipe(int position, Recipe recipe){
        recipeList.set(position, recipe);
    }

    public void deleteRecipe(int position){
        recipeList.remove(position);
    }

    public Recipe getRecipe(int position){
        Recipe recipe = recipeList.get(position);
        return recipe;
    }

    public int getSize(){
        return recipeList.size();
    }

    public ArrayList<Recipe> getRecipeList(){
        return recipeList;
    }
}

File: app/src/main/java/com/rankin/adam/cookingmaster/model/RecipeList.kt
package com.rankin.adam.cookingmaster.model

/**
 * Created by Adam on 14-Jul-18.
 */
class RecipeList {

    var recipeList: ArrayList<Recipe> = ArrayList()


    fun addRecipe(recipe: Recipe) {
        recipeList.add(recipe)
    }

    fun editRecipe(position: Int, recipe: Recipe) {
        recipeList[position] = recipe
    }

    fun deleteRecipe(recipe: Recipe) {
        recipeList.remove(recipe)
    }

    fun getRecipe(position: Int): Recipe {
        return recipeList[position]
    }

    val size: Int
        get() = recipeList.size
}